testing/src/test/java/org/motechproject/nms/testing/it/api/UserControllerBundleIT.java

```diff
@@ -5171,4 +5171,108 @@ public void verifyFT534() throws IOException, InterruptedException {
+                EntityUtils.toString(response.getEntity()));
+
+        HttpPost httpPost = createHttpPost("mobilekunji", new UserLanguageRequest(1200000000L, 123456789012346L,
+                rh.hindiLanguage().getCode()));
+
+        response = SimpleHttpClient.httpRequestAndResponse(httpPost, ADMIN_USERNAME, ADMIN_PASSWORD);
+
+        assertEquals(HttpStatus.SC_OK, response.getStatusLine().getStatusCode());
+
+        FrontLineWorker flw = frontLineWorkerService.getByContactNumber(1200000000l);
+        assertEquals(FrontLineWorkerStatus.ANONYMOUS, flw.getStatus());
+    }
+
+    /**
+     * To verift that status of flw must be set to "Active" when user call first time and
+     * its information exists in NMS DB and status as "Inactive"
+     */
+    @Ignore
+    // TODO : https://applab.atlassian.net/browse/NMS-246
+    @Test
+    public void verifyFT512() throws IOException, InterruptedException {
+    rh.delhiState();
+    rh.newDelhiDistrict();
+    rh.delhiCircle();
+    deployedServiceDataService.create(new DeployedService(rh.delhiState(),
+            Service.MOBILE_KUNJI));
+
+    FrontLineWorker flw = new FrontLineWorker("Frank Lloyd Wright", 1200000001L);
+    flw.setLanguage(rh.hindiLanguage());
+    flw.setDistrict(rh.newDelhiDistrict());
+    flw.setState(rh.delhiState());
+    flw.setStatus(FrontLineWorkerStatus.INACTIVE);
+    frontLineWorkerService.add(flw);
+
+    // invoke get user detail API
+       HttpGet httpGet = createHttpGet(true, "mobilekunji", // service
+            true, "1200000001", // callingNumber
+            false, null, // operator
+            true, rh.delhiCircle().getName(),// circle
+            true, "123456789012346" // callId
+    );
+
+    String expectedJsonResponse = createFlwUserResponseJson(rh.hindiLanguage().getCode(), // defaultLanguageLocationCode
+            rh.hindiLanguage().getCode(), // locationCode
+            new ArrayList<String>(), // allowedLanguageLocationCodes
+            0L, // currentUsageInPulses=updated
+            0L, // endOfUsagePromptCounter=updated
+            false, // welcomePromptFlag
+            -1, // maxAllowedUsageInPulses=State capping
+            2 // maxAllowedEndOfUsagePrompt
+    );
+
+    HttpResponse response = SimpleHttpClient.httpRequestAndResponse(httpGet,
+            ADMIN_USERNAME, ADMIN_PASSWORD);
+    assertEquals(HttpStatus.SC_OK, response.getStatusLine().getStatusCode());
+    assertEquals(expectedJsonResponse,
+            EntityUtils.toString(response.getEntity()));
+
+    HttpPost httpPost = createHttpPost("mobilekunji", new UserLanguageRequest(1200000001L, 123456789012346L,
+            rh.hindiLanguage().getCode()));
+
+    response = SimpleHttpClient.httpRequestAndResponse(httpPost, ADMIN_USERNAME, ADMIN_PASSWORD);
+
+    assertEquals(HttpStatus.SC_OK, response.getStatusLine().getStatusCode());
+
+    flw = frontLineWorkerService.getByContactNumber(1200000001l);
+    assertEquals(FrontLineWorkerStatus.ACTIVE, flw.getStatus());
```

Comment:
As discussed in https://applab.atlassian.net/browse/NMS-246 we don't set the user to active until we get the call details back from IMI.  So this test isn't valid.  If you update it to simulate the call details and then check the user it will pass.